git/compat.py

```diff
@@ -46,6 +47,20 @@ def mviter(d):
     def mviter(d):
         return d.itervalues()
 
+PRE_PY27 = sys.version_info < (2, 7)
+
+
+def safe_decode(s):
+    """Safely decodes a binary string to unicode"""
+    if isinstance(s, six.text_type):
+        return s
+    elif isinstance(s, six.binary_type):
+        if PRE_PY27:
+            return s.decode(defenc)  # we're screwed
```

Comment:
You're absolutely right here, this is inappropriate. Fixed in aae2a7328a4d28077a4b4182b4f36f19c953765b.